### PR TITLE
Fix ipython + visible console results in AttributeError

### LINKED ISSUE #2880: ipython + visible console results in AttributeError
## 🐛 Bug

When instantiating a napari viewer with a visible console dock from inside ipython results in `AttributeError` (seems to be something related to `ipython` itself, see below)

## To Reproduce

1. start napari from command line `napari`
2. select the console dock to be visible and close again
3. start napari from within ipython `ipython -c "import napari;napari.Viewer()"`

```
napari/_qt/qt_viewer.py in <lambda>(visible)
    173         # gets (or creates) the console when the dock console is made visible.
    174         self.dockConsole.visibilityChanged.connect(
--> 175             lambda visible: self.console if visible else None
    176         )
    177         self.dockLayerControls.visibilityChanged.connect(self._constrain_width)

AttributeError: 'QtViewer' object has no attribute 'console'
```

This does *not* happen, when
*  the console dock is not visible
*  using `python` instead of `ipython`, i.e. `python -c "import napari;napari.Viewer()"` works

## Environment
```
napari: 0.4.9
Platform: macOS-10.15.7-x86_64-i386-64bit
System: MacOS 10.15.7
Python: 3.8.0 (default, Nov 6 2019, 15:49:01) [Clang 4.0.1 (tags/RELEASE_401/final)]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.20.3
SciPy: 1.6.3
Dask: 2021.06.0
VisPy: 0.6.6

OpenGL:
- GL version: 2.1 ATI-3.10.18
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1792x1120, scale 2.0

Plugins:
- console: 0.0.3
- scikit-image: 0.4.9
- svg: 0.1.5
```

## Additional context

happens too on linux and python 3.8/3.9

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
fixes #2880

@Carreau, if you could have a quick look that would be great.  Not 100% sure whether `console.push` is doing the same thing as the `user_ns` update that was added in #2677 ... but moving or commenting these two lines is sufficient to fix this bug (though, tbh, I'm not 100% sure why)

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
LGTM! But i'll wait till @Carreau has had a chance to weigh in before merge

**Comment by Carreau** [comment]
Yep +1, that does about the same as before with some extra safeguards.

**Review by Carreau — APPROVED** [review]

**Pull request merged**
